lib/helpers.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Fields.Helpers do
+  @doc """
+  Hash a string, or a value that implements the String.Chars protocol, using
+  Argon2. Argon2 is a strong but slow hashing function, so is recommended
+  for passwords.
+  """
+  @spec hash(atom(), String.Chars.t()) :: String.t()
+  def hash(:argon2, value) do
+    Argon2.Base.hash_password(to_string(value), Argon2.gen_salt(), [{:argon2_type, 2}])
+  end
+
+  @doc """
+  Hash a string, or a value that implements the String.Chars protocol, using
+  sha256. sha256 is fast, but not as strong as Argon2,
+  so it is not recommended for hashing passwords.
+  """
+  @spec hash(atom(), String.Chars.t()) :: String.t()
+  def hash(:sha256, value) do
+    :crypto.hash(:sha256, to_string(value) <> get_salt(to_string(value)))
+  end
+
+  defp get_salt(value) do
+    secret_key_base = Application.get_env(:fields, Fields)[:secret_key_base]
```

Comment:
@Danwhy does this line imply that the `:secret_key_base` will be sourced from the Fields Module config?
Or will it be a `:fields` key in the `config.exs` file of the project _using_ Fields?

Reply:
It should come from the application using Fields. I will have to double check, but I don't think another application will be able to access the Fields module config, as I don't think it's part of the compiled code.